highs/mip/feasibilityjump.hh

```diff
@@ -0,0 +1,775 @@
+#include <algorithm>
+#include <cassert>
+#include <climits>
+#include <cmath>
+#include <functional>
+#include <numeric>
+#include <random>
+#include <vector>
+
+#define FJ_LOG_PREFIX "Feasibility Jump: "
+
+// TIP: clang-format the reference feasibilityjump.hh before diffing with this
+namespace external_feasibilityjump {
+
+enum RowType {
+  Equal,
+  Lte,
+  Gte,
+};
+
+enum VarType { Continuous, Integer };
+
+enum CallbackControlFlow {
+  Terminate,
+  Continue,
+};
+
+struct FJStatus {
+  size_t totalEffort;
+  size_t effortSinceLastImprovement;
+  int numVars;
+  double solutionObjectiveValue;
+  double* solution;
+};
+
+double violationTolerance = 1.0e-5;
```

Comment:
either this can be const or we have a threading issue if HiGHS is used as a library and asked to solve several instances in parallel.

Reply:
Thanks. I guess it can be const, but if it's never changed, how it can it cause problems in a multi-threaded application?

Reply:
if it is not written to it is not an issue. I guess it's a matter of personal preference. I find it nice to let the compiler validate the assumption that it is not written to.
In this case the compiler disagrees due to HighsFeasibilityJump.cpp line 31 and 32.

Reply:
Thanks, I see the issue now. @galabovaa and @BenChampion , how can this most easily be ifxed?

Reply:
The easiest potential fix I see at the moment is to make `violationTolerance` a instance variable of `Problem`, `equalityTolerance` a instance variable of both `Problem` and `JumpMove`, and `eq` a member function of both `Problem` and `JumpMove`. Then we can pass both `violationTolerance`  and `equalityTolerance`  through the `FeasibilityJumpSolver` constructor to its `Problem` and `JumpMove` members.

It doesn't seem particularly elegant, but I think it would do the job?

Reply:
Sounds alright to me, we need some way of keeping the values attached to the solver instances and this sounds OK

Reply:
Cool, I've updated the PR. (I added an additional tolerance argument to `eq` meaning I don't have to make it a member function anywhere.)